src/outputs/history.cpp

```diff
@@ -49,68 +49,111 @@ void HistoryOutput::WriteOutputFile(Mesh *pm, ParameterInput *pin, SimTime *tm,
   if (signal == SignalHandler::OutputSignal::now) {
     return;
   }
-  std::vector<std::string> all_labels = {};
-  std::vector<Real> all_results = {};
+
+  std::map<UserHistoryOperation, std::vector<Real>> results;
+  std::map<UserHistoryOperation, std::vector<std::string>> labels;
+  std::vector<UserHistoryOperation> ops = {
+      UserHistoryOperation::sum, UserHistoryOperation::max, UserHistoryOperation::min};
+  for (auto &op : ops) {
+    results[op] = {};
+    labels[op] = {};
+  }
+
+  // If "packages" not provided, output history for all packages
+  auto &requested_packages = output_params.packages;
+  Dictionary<std::shared_ptr<StateDescriptor>> packages;
```

Comment:
You could use a `Packages_t` here but that might be overkill

Reply:
Looking at `Packages_t` I do think it would be overkill here.